addons/auth/addon/authenticators/base.js

```diff
@@ -58,10 +58,14 @@ export default class BaseAuthenticator extends SimpleAuthBaseAuthenticator {
    */
   async validateToken(token, tokenID) {
     const tokenValidationURL = this.buildTokenValidationEndpointURL(tokenID);
-    const response = await fetch(tokenValidationURL, {
-      method: 'get',
-      headers: { Authorization: `Bearer ${token}` },
-    });
+    // Note: waitForPromise is needed to provide the necessary integration with @ember/test-helpers
```

Comment:
Is this needed because there are there certain tests that relied on waiting for the ember run loop?

Reply:
yeah without this some of the authentication tests in [ui/admin/tests/acceptance/authentication-test.js](https://github.com/hashicorp/boundary-ui/blob/main/ui/admin/tests/acceptance/authentication-test.js) were failing. The helper lets internals of test helpers know that there is a pending request in the run loop so that it waits for it to settle.

Reply:
Yeah that's what I assumed, not ideal since I'm really not a fan of code that's just added for testing but I also haven't looked at these tests before 😢

Reply:
agreed, tho i didn't find an alternative. Feel free to suggest any ideas to explore.

Reply:
I think it's fine for now, I'm not sure why we were using `ember-fetch` instead of native anyways so that might have been the reason in the first place